WriteHDLUtils.py

```diff
@@ -255,106 +258,44 @@ def getHLSMemoryClassName(module):
     else:
         raise ValueError(module.mtype + " is unknown.")
 
-def labelConnectedMemoryArrays(proc_list):
-    """
-    # label those memories that will be constructed in an array
-    # (if these scripts end up generating the HLS top levels, will this function
-    # be needed? That is, even if the templated HLS function has arrays on the
-    # interface, the top-level could still be individual memories, although you'd
-    # still have to pass those individual memories to the templated block as an array
-    """
-    for aProcModule in proc_list:
-
-        if aProcModule.mtype == 'TrackletCalculator':
-            # stubpair and allstub memories are arrays
-            # they are all inputs
-            for inmem,inport in zip(aProcModule.upstreams,aProcModule.input_port_names):
-                if 'stubpair' in inport:
-                    inmem.userlabel = 'stubPairs_'+aProcModule.inst
-                elif 'innerallstubin' in inport:
-                    inmem.userlabel = 'innerStubs_'+aProcModule.inst
-                elif 'outerallstubin' in inport:
-                    inmem.userlabel = 'outerStubs_'+aProcModule.inst
-
-        # add other processing steps here if they have array inputs/outputs
-        #elif aProcModule.mtype == '':
-
 def getListsOfGroupedMemories(aProcModule):
     """
     # Get a list of memories and a list of ports for a given processing module
     # The memories are further grouped in a list if they are expected to be
     # constructed and passed to the processing function as an array
-
-    # add array name to 'userlabel' of the connected memory module
     """
-    labelConnectedMemoryArrays([aProcModule])
-
     memList = list(aProcModule.upstreams + aProcModule.downstreams)
     portList = list(aProcModule.input_port_names + aProcModule.output_port_names)
     # sort?
 
     newmemList = []
     newportList = []
-    arraycontainer_dict = {}
 
     for memory, portname in zip(memList, portList):
-        if not memory.userlabel:  # default is '', i.e. no array label added
-            newmemList.append(memory)
-            newportList.append(portname)
-        else:
-            arrayname = memory.userlabel
-            if arrayname not in arraycontainer_dict:
-                arraycontainer_dict[arrayname] = [memory]
-            else:
-                arraycontainer_dict[arrayname].append(memory)
-    # add back memory arrays
-    for arrayname in arraycontainer_dict:
-        newmemList.append(arraycontainer_dict[arrayname])
-        newportList.append(arrayname)
+        newmemList.append(memory)
+        newportList.append(portname)
```

Comment:
I might be stupid but what's the point of this? :-) Can't you just return memList portList?

Reply:
You're correct. This was historic code, which is no longer needed. Now removed.